File: misc/aooswd/ex/singleton/singleton.cpp
class Singleton
{
public:
  static Singleton* Instance ();
protected:
  Singleton ();
private:
  static Singleton* _instance;
};

Singleton* Singleton::_instance = 0;
Singleton* Singleton::Instance ()
{
  if (_instance == 0)
    _instance = new Singleton;
  return _instance;
}

class Singleton
{
public:
  static void Register (char* name, Singleton*);
  static Singleton* Instance ();
protected:
  static Singleton* Lookup (const char* name);
private:
  static Singleton* _instance;
  static List<NameSingletonPair>* _registry;
};

File: misc/aooswd/ex/singleton/singleton.java
public class Singleton
{
	public static Singleton Instance ()
	{
		if (_instance == null)
  			_instance = new Singleton();
 		return _instance;
	}
	protected Singleton (){}
	private static Singleton _instance = null;
}

public class Singleton
{
	public static void Register (String name, Singleton singleton);
	public static Singleton Instance ();
	protected static Singleton Lookup (String name);
	private static Singleton _instance;
	private static Collection _registry;
};
